bowtie/_core.py

```diff
@@ -538,6 +543,45 @@ def supports(self, *dialects: Dialect) -> bool:
         """
         return self.info.dialects.issuperset(dialects)
 
+    async def get_versions(self) -> frozenset[zip[tuple[str, ...]]]:
+        from github3 import GitHub  # type: ignore[reportMissingTypeStubs]
+        from github3.exceptions import (  # type: ignore[reportMissingTypeStubs]
+            GitHubError,
+        )
+        from github3.models import (  # type: ignore[reportMissingTypeStubs]
+            GitHubCore,
+        )
+
+        url = CONTAINER_PACKAGES_API / self.id / "versions"
+        versions: Set[str] = set()
+
+        pages: list[GitHubCore] = []
+        try:
+            gh = GitHub(token=os.environ.get("GITHUB_TOKEN", ""))
+            pages = gh._iter(count=-1, url=str(url), cls=GitHubCore)  # type: ignore[reportPrivateUsage]
+        except GitHubError:
+            pass
+
+        for page in pages:
+            try:
+                tags = cast(
+                    Iterable[str],
+                    page.as_dict()["metadata"]["container"]["tags"],
+                )
+            except KeyError:
+                continue
+            else:
+                versions.update([tag for tag in tags if "." in tag])
+
+        if not versions and self.info.version:
```

Comment:
This seems like we should do it always, rather than just if we didn't get a response -- if the current container is for a version not in the list, obviously that version still exists right :)?